tests/PHPStan/Analyser/data/json-decode/narrow_type_with_force_array.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Analyser\JsonDecode;
+
+use function PHPStan\Testing\assertType;
+
+$value = json_decode('true', true);
+assertType('true', $value);
+
+$value = json_decode('1', true);
+assertType('1', $value);
+
+$value = json_decode('1.5', true);
+assertType('1.5', $value);
+
+$value = json_decode('false', true);
```

Comment:
Please also test second argument `false` and also please test `null` with `JSON_OBJECT_AS_ARRAY`: 

> When true, JSON objects will be returned as associative arrays; when false, JSON objects will be returned as objects. When null, JSON objects will be returned as associative arrays or objects depending on whether JSON_OBJECT_AS_ARRAY is set in the flags.

Reply:
I'm sure how the `JSON_OBJECT_AS_ARRAY` exactly works. I haven't use `json_decode` in last couple of years.

What is the exptected type? It seems the result with `JSON_OBJECT_AS_ARRAY` is same as 2nd parameter `true`
https://3v4l.org/ffeIM

Reply:
Yes, but we need to check that PHPStan understands it as well when called as `json_decode($input, null, JSON_OBJECT_AS_ARRAY)`. And also the other behaviour when called without it.

Reply:
I've added separate `json_object_as_array` to cover it.

Maybe one more test in case of combination of constants.